datafusion/core/src/execution/context/mod.rs

```diff
@@ -813,24 +815,26 @@ impl SessionContext {
             input,
             or_replace,
             definition,
+            temporary,
         } = cmd;
 
         let view = self.table(name.clone()).await;
 
-        match (or_replace, view) {
-            (true, Ok(_)) => {
+        match (or_replace, view, temporary) {
+            (_, _, true) => exec_err!("'TEMPORARY' view not defined'"),
```

Comment:
i am unsure if the `exec_err!` here is appropriate, should the temporary semantics bleed into the execution?

Reply:
Similar to above. It would be simpler to check the `temporary` flag outside the match expression, and it would be more appropriate to use `not_impl_err`.